stdlib/http/src/main/ballerina/http/http_client.bal

```diff
@@ -20,17 +20,16 @@
 # generic HTTP actions implementation.
 #
 # + url - The URL of the remote HTTP endpoint
-# + config - The configurations associated with the HttpCaller
-public type HttpCaller client object {
```

Comment:
Wasn't `HttpCaller` what we used for the service to interact with the client?

Reply:
That is the Caller object which is used in the services and it has no relation with this HttpClient. In order to avoid that confusion, we thought of replacing the HttpCaller with HttpClient. HttpClient is basically the simple client which wraps the java client connector. And it is the base object for other clients(Caching, Retry,...etc)